packages/js/src/settings/store/pages.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable camelcase, complexity */
+import { createEntityAdapter, createSelector, createSlice } from "@reduxjs/toolkit";
+import apiFetch from "@wordpress/api-fetch";
+import { buildQueryString } from "@wordpress/url";
+import { map, trim } from "lodash";
+import { ASYNC_ACTION_NAMES, ASYNC_ACTION_STATUS } from "../constants";
+
+const pagesAdapter = createEntityAdapter();
+
+export const FETCH_PAGES_ACTION_NAME = "fetchPages";
+// Global abort controller for this reducer to abort requests made by multiple selects.
+let abortController;
+
+/**
+ * @param {Object} queryData The query data.
+ * @returns {Object} Success or error action object.
+ */
+export function* fetchPages( queryData ) {
```

Comment:
I don't think this export is needed (probably not in users either). We should only use it as an action through WP data, i.e. why you have in `pageActions` below.